lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -66,14 +72,52 @@ def setup_encrypter(options = {})
                                 monitoring_io: mongocryptd_client_monitoring_io,
                               )
 
+        unless @encryption_options[:mongocryptd_bypass_spawn] || !mongocryptd_client_monitoring_io
+          self.spawn_mongocryptd
+        end
+
         # TODO: use all the other options for auto-encryption/auto-decryption
       end
 
+      # Spawn a new mongocryptd process using the mongocryptd_spawn_path
+      # and mongocryptd_spawn_args passed in through the extra auto
+      # encrypt options. Stdout and Stderr of this new process are written
+      # to /dev/null.
+      #
+      # @return [ Integer ] The process id of the spawned process
+      def spawn_mongocryptd
+        @mongocryptd_pid = Process.spawn(
+                            @encryption_options[:mongocryptd_spawn_path],
+                            *@encryption_options[:mongocryptd_spawn_args],
+                            [:out, :err]=>'/dev/null'
+                          )
+      end
+
+      # Abort the spawned mongocryptd process. Will not raise an error,
+      # even if the process has already been terminated.
+      #
+      # @return [ true ] Always true
+      def kill_mongocryptd
+        return true if @mongocryptd_pid.nil?
+
+        begin
+          Process.kill('ABRT', @mongocryptd_pid)
+          Process.wait(@mongocryptd_pid)
+        rescue Errno::ESRCH
+         log_warn("There was no mongocrypd process running at #{@mongocryptd_pid}.")
+        end
+
+        @mongocryptd_pid = nil
+
+        true
+      end
+
       # Close the resources created by the AutoEncrypter
       #
       # @return [ true ] Always true
       def teardown_encrypter
         @mongocryptd_client.close if @mongocryptd_client
+        kill_mongocryptd
```

Comment:
I thought the daemon process would live forever and just the client would be closed here.

Reply:
You're completely right. From the spec:
> The mongocryptd process is responsible for self terminating after idling for a time period.

I'll remove everything having to do with terminating it.